apps/src/code-studio/header.js

```diff
@@ -173,21 +190,72 @@ function shareProject() {
 
     ReactDOM.render(
       <Provider store={getStore()}>
-        <ShareDialog
+        <ConnectedShareDialog
           i18n={i18n}
           icon={appOptions.skin.staticAvatar}
           shareUrl={shareUrl}
           isAbusive={dashboard.project.exceedsAbuseThreshold()}
+          isOwner={dashboard.project.isOwner()}
+          isPublished={dashboard.project.isPublished()}
           channelId={dashboard.project.getCurrentId()}
           appType={appType}
           onClickPopup={popupWindow}
           // TODO: Can I not proliferate the use of global references to Applab somehow?
           onClickExport={window.Applab ? window.Applab.exportApp : null}
+          onShowPublishDialog={showPublishDialog}
+          onUnpublish={unpublishProject}
           canShareSocial={canShareSocial}
         />
       </Provider>,
       dialogDom
     );
+
+    getStore().dispatch(showShareDialog());
+  });
+}
+
+const ConnectedPublishDialog = connect(state => ({
+  isOpen: state.header.publishDialog.isOpen,
+  isPublishPending: state.header.publishDialog.isPublishPending,
+  projectId: state.header.publishDialog.projectId,
+  projectType: state.header.publishDialog.projectType,
+}), dispatch => ({
+  onClose: () => dispatch(hidePublishDialog()),
+}))(PublishDialog);
+
+function showPublishDialog() {
```

Comment:
PublishDialog moved into ShareDialog in https://github.com/code-dot-org/code-dot-org/pull/16593/commits/cd477295c060ef0fe749b126cc8bb763020100be and https://github.com/code-dot-org/code-dot-org/pull/16593/commits/bca7c5d0dedc24a9f04e5159cb51754b68b16598